src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplateBuilder.java

```diff
@@ -644,4 +649,46 @@ private List<Long> parseSupplementalGroupList(String gids) {
         }
         return Collections.unmodifiableList(builder);
     }
+
+    /**
+     * <p>
+     * {@link PodDecorator} allowing to inject in {@code jnlp} containers definition a {@code securityContext} definition allowing to use the
+     * {@code restricted} <a href="https://kubernetes.io/docs/concepts/security/pod-security-standards/">Pod Security Standard</a>.
+     * </p>
+     * <p>
+     * See <a href="https://issues.jenkins.io/browse/JENKINS-71639">JENKINS-71639</a> for more details.
+     * </p>
+     */
+    @Extension
+    public static class RestrictedPssSecurityContextInjector implements PodDecorator {
+
+        @NonNull
+        @Override
+        public Pod decorate(@NonNull KubernetesCloud kubernetesCloud, @NonNull Pod pod) {
+            if (kubernetesCloud.isRestrictedPssSecurityContext()) {
+                Optional<Container> maybeJNLP = pod.getSpec().getContainers().stream().filter(container -> JNLP_NAME.equals(container.getName())).findFirst();
+
+                maybeJNLP.ifPresentOrElse(jnlp -> {
+                    if (jnlp.getSecurityContext() != null) {
+                        LOGGER.warning(() -> "Overriding the existing JNLP container Security Context due to the configured restricted PSP injection");
+                    } else {
+                        LOGGER.fine(() -> "Injecting restricted PSP configuration in the JNLP container security context");
+                    }
+                    jnlp.setSecurityContext(new SecurityContextBuilder()                //
+                                                .withAllowPrivilegeEscalation(false)    //
+                                                .withNewCapabilities()                  //
+                                                    .withDrop("ALL")                    //
+                                                .endCapabilities()                      //
+                                                .withRunAsNonRoot()                     //
+                                                .withNewSeccompProfile()                //
+                                                    .withType("RuntimeDefault")         //
+                                                .endSeccompProfile()                    //
+                                            .build());                                  //
+                }, () -> {
+                    LOGGER.severe(() -> "Cannot find the jnlp container to configure its security when it is expected to be present.");
+                });
```

Comment:
or 
```suggestion
                }, () -> LOGGER.severe("Cannot find the jnlp container to configure its security when it is expected to be present."));
```
(though this message with no context is going to be useless)

Reply:
Replaced it with an `IllegalStateException` as mentioned by Vincent